numpy incompatibility

We have non np 1.6 compatible code in the gdk backend which we do not test on travis.

Issue reported by @cgohlke 

http://matplotlib.1069221.n5.nabble.com/Matplotlib-devel-First-release-candidate-for-the-1-5-0-series-tp46156p46166.html

http://stackoverflow.com/questions/28499580/matplotlib-import-error-backend-gdk-soundefined-symbol-pyarray-setbaseobjec

The code in question went in in 
https://github.com/matplotlib/matplotlib/pull/3665

which was in response to discussion at

http://thread.gmane.org/gmane.comp.python.matplotlib.general/34771/focus=34771

Options here are: 

 - fix code to address both issues (no idea how hard that would be)
 - up the minimal numpy until it works

I am in favor of bumping the numpy version, but do not feel strongly.

Unfortunately it's not a compilation error: it's just
```
    src/_backend_gdk.c:55:5: warning: implicit declaration of function ‘PyArray_SetBaseObject’ [-Wimplicit-function-declaration]
         PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
         ^
```
But `import matplotlib.backends.backend_gdk` triggers the error for me.

--- a/src/_backend_gdk.c
+++ b/src/_backend_gdk.c
@@ -52,7 +52,15 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
 
     /* the array holds a ref to the pixbuf pixels through this wrapper*/
     Py_INCREF(py_pixbuf);
-    PyArray_SetBaseObject(array, (PyObject *)py_pixbuf);
+#if NPY_API_VERSION >= 0x00000007
+    if (PyArray_SetBaseObject(array, (PyObject *)py_pixbuf) == -1) {
+        Py_DECREF(py_pixbuf);
+        Py_DECREF(array);
+        return NULL;
+    }
+#else
+    PyArray_BASE(array) = py_pixbuf;
+#endif
     return PyArray_Return(array);
 }
 

--- a/src/_backend_gdk.c
+++ b/src/_backend_gdk.c
@@ -59,7 +59,7 @@ static PyObject *pixbuf_get_pixels_array(PyObject *self, PyObject *args)
         return NULL;
     }
 #else
-    PyArray_BASE(array) = py_pixbuf;
+    PyArray_BASE(array) = (PyObject *) py_pixbuf;
 #endif
     return PyArray_Return(array);
 }

